pkg/stream/integration/pg_pg_integration_transformer_test.go

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	pglib "github.com/xataio/pgstream/internal/postgres"
+	"github.com/xataio/pgstream/pkg/stream"
+)
+
+type transformerTestTableRow struct {
+	id             int
+	name           string
+	lastName       string
+	email          string
+	address        string
+	age            int
+	totalPurchases float64
+	customerID     string
+	birthDate      time.Time
+	isActive       bool
+	createdAt      int64
+	updatedAt      time.Time
+	gender         string
+}
+
+var createTableQuery = `CREATE TABLE %s(
+	id serial primary key,
+	name text,
+	last_name varchar(255),
+	email varchar(255),
+	address text,
+	age integer,
+	total_purchases double precision,
+	customer_id uuid,
+	birth_date date,
+	is_active bool,
+	created_at bigint,
+	updated_at timestamp with time zone,
+	gender varchar(255));`
+
+func Test_PostgresToPostgres_Transformer(t *testing.T) {
+	if os.Getenv("PGSTREAM_INTEGRATION_TESTS") == "" {
+		t.Skip("skipping integration test...")
+	}
+
+	cfg := &stream.Config{
+		Listener:  testPostgresListenerCfg(),
+		Processor: testPostgresProcessorCfgWithTransformer(pgurl),
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	runStream(t, ctx, cfg)
+
+	testTable := "pg2pg_integration_transformer_test"
+	execQuery(t, ctx, fmt.Sprintf(createTableQuery, testTable))
+
+	targetConn, err := pglib.NewConn(ctx, targetPGURL)
```

Comment:
I just realised the integration tests don't close the conns. While it's not a big deal, since the postgres instance is tear down after the test, it's probably good practice to just defer close them. If you don't mind adding it here and in the `execQuery` part please 🙏